### PR TITLE
feat(nexus): enable docker-based wheel building for nexus

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
In this PR:
- Introduced `nox` to streamline building nexus and installing it. To build the nexus wheel for the current platform and install it into the active python environment, run:
```shell
nox -s build-nexus install-nexus 
```
- Under the hood, use `python -m build` instead of implicitly using `setup.py bdist_wheel` to build the wheels. See https://pypa-build.readthedocs.io/en/stable/
- When building linux-amd64 wheels on a Mac, use docker to make cgo+go-nvml happy.

Under the hood:
To build the nexus wheel only for (a) specific platform(s), run:
```shell
python -m build -w -n ./nexus --config-setting=--build-option=--nexus-build=darwin-arm64,linux-amd64
```
(yes, it is this ugly, but it works -- spent like an hour trying to dig this one out)

To build the nexus wheel for all platforms, run:
```shell
python -m build -w -n ./nexus
```

To install the nexus wheel, run:
```shell
pip install --force-reinstall ./nexus/dist/wandb_core-*-py3-none-any.whl
```

### <samp>🤖 Generated by Copilot at ad2c9e1</samp>

This pull request adds and updates some files to enable building and installing the nexus binary, a dependency of wandb-core, for different platforms using docker and nox. It modifies `nexus/setup.py`, `tox.ini`, and `requirements_build.txt`, and adds `nexus/scripts/build/Dockerfile` and `noxfile.py`.

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at ad2c9e1</samp>

> _To build nexus wheels with ease_
> _We added some docker and nox_
> _The setup.py file got a fix_
> _To handle cgo and tox_
> _Now we can test wandb-core with these_

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -155,7 +156,10 @@ allowlist_externals=
 commands=
     mkdir -p test-results
     # install nexus as wandb-core, requires go to be installed on the system
-    pip install ./nexus
+    # todo: this is for the CI, not to compile for the platforms we're not going to use
+    python -m build -w -n ./nexus --config-setting=--build-option=--nexus-build=linux-amd64
+    # todo: do not pin to a specific version, or bump automatically
+    pip install --force-reinstall ./nexus/dist/wandb_core-0.0.1a3-py3-none-any.whl
```
Comment (raubitsj) [inline tox.ini:162]: i think this will be a bit fragile even in the short term.  maybe just a wildcard here for the version for now?

**Review by raubitsj — APPROVED** [review]
nice

File: tox.ini (hunk 1 above)
Comment (dmitryduev) [reply tox.ini:162]: i had a wildcard there, but then tox forces you to do /bin/bash -c ..., which you need to whitelist and also pass the env (maybe? didn't check)

Comment (dmitryduev) [reply tox.ini:162]: will think of a better way, i agree that it's fragile

**Review by kptkin — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (dmitryduev) [reply tox.ini]: after a long battle with tox, introduced nox as a solution. so far loving it! (one can do what I wanted with tox, it's just so much cleaner, easier, and out-of-the-boxy with nox)

Comment (dmitryduev) [reply tox.ini]: check out how nice the logs look: https://app.circleci.com/pipelines/github/wandb/wandb/25981/workflows/069f22ee-d3c7-4399-badb-3fb0767d77b9/jobs/742972?invite=true#step-105-6301_93

**Pull request merged**